examples/direct_fidelity_estimation.py

```diff
@@ -29,15 +31,19 @@ def build_circuit():
 
 def compute_characteristic_function(circuit: cirq.Circuit,
                                     P_i: Tuple[cirq.Gate, ...],
-                                    qubits: List[cirq.Qid],
-                                    noise: Optional[cirq.NoiseModel]):
+                                    qubits: List[cirq.Qid], physical: bool,
+                                    simulator: cirq.DensityMatrixSimulator,
+                                    density_matrix: Optional[np.ndarray]):
     n_qubits = len(P_i)
     d = 2**n_qubits
 
-    simulator = cirq.DensityMatrixSimulator()
-    # rho or sigma in https://arxiv.org/pdf/1104.3835.pdf
-    density_matrix = cast(cirq.DensityMatrixTrialResult,
-                          simulator.simulate(circuit)).final_density_matrix
+    if physical:
+        assert density_matrix is None
```

Comment:
this function is mostly in an if statement and one of the arguments is unused. consider splitting into two functions

Reply:
I have a compute_characteristic_function() function that only does one thing. Would you like me to put the code that uses PauliSumCollector in its own function too? Right now, it's not because it is only called once.